plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -143,6 +144,15 @@ func (s *serviceAccount) Admit(a admission.Attributes) (err error) {
 		return nil
 	}
 
+	updateInitialized, err := util.IsUpdatingInitializedObject(a)
+	if err != nil {
+		return err
+	}
+	if updateInitialized {
```

Comment:
After the pod is initialized, `pod.spec.serviceaccountname/imagepullsecrets/volumes` are immutable, so the serviceaccount admission controller doesn't need to take any action if the request is updating an already initialized pod.